src/main/java/org/apache/maven/plugins/checkstyle/CheckstyleViolationCheckMojo.java

```diff
@@ -72,7 +72,7 @@
  * @author <a href="mailto:joakim@erdfelt.net">Joakim Erdfelt</a>
  * @version $Id$
  */
-@Mojo( name = "check", defaultPhase = LifecyclePhase.VERIFY, requiresDependencyResolution = ResolutionScope.TEST,
+@Mojo( name = "check", defaultPhase = LifecyclePhase.VERIFY, requiresDependencyResolution = ResolutionScope.NONE,
```

Comment:
I think a good comprimise would be to switch to "COMPILE" and not to "NONE"

This way we will save resources (The case in MCHECKSTYLE-353 is about test dependencies ?) and it will save the case in which you are bundling checkstyle configuration on the classpath

Does this solution work for you @zregvart  ?

Reply:
Thanks for proposing this. I'll measure the execution times with empty local Maven repository on Apache Camel and report my findings.